plutus-ledger-api/src/PlutusLedgerApi/V1/Value.hs

```diff
@@ -251,6 +254,15 @@ valueOf (Value mp) cur tn =
             Nothing -> 0
             Just v  -> v
 
+{-# INLINABLE currencySymbolValueOf #-}
+currencySymbolValueOf :: Value -> CurrencySymbol -> Integer
+currencySymbolValueOf (Value mp) cur = case Map.lookup cur mp of
+    Nothing     -> 0
+    Just tokens ->
+        -- This is more efficient than `PlutusTx.sum (Map.elems tokens)`, because
+        -- the latter materializes the intermediate result of `Map.elems tokens`.
```

Comment:
Are you sure it doesn't get inlined away? Would be good to add the golden test first, then redefine the function. Anyway, not important.

Reply:
Yes, with `sum . Map.elems`:

```
-({cpu: 26943308
-| mem: 104124})

+({cpu: 30347308
+| mem: 118924})
```

And it is not supposed to be inlined away in the first place.

Reply:
Thanks! 

> And it is not supposed to be inlined away in the first place.

Yeah, we're not doing fusion/deforestation, in Haskell it _could_ optimize away.